core/src/main/java/org/jobrunr/storage/sql/common/DatabaseCreator.java

```
@@ -213,7 +253,136 @@ private TablePrefixStatementUpdater getStatementUpdater(String tablePrefix, Conn
+            lockUpdateScheduler.scheduleAtFixedRate(this::updateMigrationsTableLock, 5, 5, TimeUnit.SECONDS);
+        }
+
+        private void removeMigrationsTableLock() {
+            LOGGER.debug("Removing lock on migrations table...");
+            lockUpdateScheduler.shutdown();
+            try (final Connection conn = getConnection(); final Transaction tran = new Transaction(conn)) {
+                removeLock(conn);
+                tran.commit();
+            } catch (Exception e) {
+                throw shouldNotHappenException(new IllegalStateException("Error removing lock from migrations table", e));
+            }
+        }
+
+        private void updateMigrationsTableLock() {
+            LOGGER.debug("Updating lock on migrations table...");
+            try (final Connection conn = getConnection(); final Transaction tran = new Transaction(conn)) {
+                updateLock(conn);
+                tran.commit();
+            } catch (Exception e) {
+                throw shouldNotHappenException(new IllegalStateException("Error removing lock from migrations table", e));
+            }
+        }
+
+        private void waitUntilMigrationsAreDone() {
+            LOGGER.info("Waiting for database migrations to finish...");
+            try {
+                while (isMigrationsTableLocked()) {
+                    sleep(2000);
+                }
+            } catch (InterruptedException e) {
+                LOGGER.warn("Server was stopped before all migrations tables were finished.");
+                Thread.currentThread().interrupt();
+            } catch (SQLException e) {
+                throw shouldNotHappenException(e);
+            } catch (Exception e) {
+                LOGGER.error("Error waiting for database migrations to finish. Manually review your database migrations in the jobrunr_migrations table and then delete the migration lock entry with id '{}' before trying again.", TABLE_LOCKER_UUID, e);
```

Should throw an exception to avoid BackgroundJobServer starting